editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -326,6 +326,8 @@ void CanvasItemEditor::_unhandled_key_input(const Ref<InputEvent> &p_ev) {
 	if (k->get_control())
 		return;
 
+	snap_key_enabled = Input::get_singleton()->is_key_pressed(KEY_CONTROL);
```

Comment:
No need to create a private variable handled in _unhandled_key_input here. `Input::get_singleton()->is_key_pressed(KEY_CONTROL)` can be called directly into the snap_angle or snap_point function.

Reply:
I just did more performance wise way to handle it(as I did for 3D in #16890), and its also uses in _draw_rulers

Reply:
In any case that's just premature optimization. Calling Input::get_singleton()->is_key_pressed() is a very fast processing.
Once again you're added 16 lines of code for a change that could be done by adding two lines.
Please keep things simple, the more we add complexity to the code the harder it gets to maintain.

Reply:
Ok